src/Components/Web/src/Routing/NavLink.cs

```diff
@@ -124,9 +132,24 @@ private bool ShouldMatch(string currentUriAbsolute)
             return true;
         }
 
+        if (_disableMatchAllIgnoresLeftUriPart)
+        {
+            return false;
+        }
+
+        string uriWithoutQueryAndFragment = GetUriIgnoreQueryAndFragment(currentUriAbsolute);
+        if (Match == NavLinkMatch.All
+            && EqualsHrefExactlyOrIfTrailingSlashAdded(uriWithoutQueryAndFragment))
+        {
+            return true;
+        }
+
         return false;
     }
 
+    private static string GetUriIgnoreQueryAndFragment(string uri) =>
+        new Uri(uri).GetLeftPart(UriPartial.Path);
```

Comment:
Can we operate directly on the string and avoid the allocations here?

https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/Routing/SupplyParameterFromQueryValueProvider.cs#L87-L99